tck/src/main/java/org/eclipse/microprofile/concurrency/tck/ManagedExecutorTest.java

```diff
@@ -1230,6 +1230,65 @@ public void shutdownPreventsAdditionalSubmits() throws ExecutionException, Inter
         }
     }
 
+    /**
+     * Verify that the ManagedExecutor.Builder can be used to create multiple ManagedExecutors with 
+     * different configured contexts.
+     */
+    @Test
+    public void reuseManagedExecutorBuilder() throws ExecutionException, InterruptedException, TimeoutException {
+        ManagedExecutor.Builder builder = ManagedExecutor.builder()
+                .propagated()
+                .cleared(Buffer.CONTEXT_NAME);
+        
+        ManagedExecutor clearingExecutor = builder.build();
+
+        builder.propagated(Buffer.CONTEXT_NAME);
+        try {
+            builder.build();
+            Assert.fail("ManagedExecutor.Builder.build() should throw an IllegalStateException for set overlap between propagated and cleared");
+        }
+        catch (IllegalStateException ISE) {
+            // test passes
+        }
+
+        builder.cleared();
+        ManagedExecutor propagatingExecutor = builder.build();
+
+        try {
+            // Set non-default value
+            Buffer.set(new StringBuffer("reuseBuilder-test-buffer-A"));
+
+            Future<Void> clearedFuture = clearingExecutor.completedFuture(1).thenRun(() -> {
+                Assert.assertEquals(Buffer.get().toString(), "",
+                        "Context type that is configured to be cleared was not cleared.");
+
+                Buffer.set(new StringBuffer("reuseBuilder-test-buffer-B"));
+            });
+
+            Future<Void> propagatedFuture = propagatingExecutor.completedFuture(1).thenRun(() -> {
+                Assert.assertEquals(Buffer.get().toString(), "reuseBuilder-test-buffer-A",
+                        "Context type was not propagated to contextual action.");
```

Comment:
This task should use thenRunAsync instead of running it inline where the context to propagate will already be on the test's main thread.

Reply:
Good point. Updating...